firebase-firestore/ktx/src/androidTest/java/com/google/firebase/firestore/DocumentIdIntegrationTest.kt

```diff
@@ -0,0 +1,87 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.firestore
+
+import com.google.common.truth.Truth.assertThat
+import com.google.firebase.firestore.ktx.annotations.KDocumentId
+import com.google.firebase.firestore.ktx.serialization.setData
+import com.google.firebase.firestore.testutil.testCollection
+import com.google.firebase.firestore.testutil.waitFor
+import kotlinx.serialization.Contextual
+import kotlinx.serialization.Serializable
+import org.junit.Test
+
+class DocumentIdIntegrationTest {
+
+    @Test
+    fun encoding_DocumentReference_is_supported() {
+        val docRefKotlin = testCollection("ktx").document("123")
+        val docRefPOJO = testCollection("pojo").document("456")
+
+        @Serializable
+        data class DocumentIdOnDocRefField(@Contextual var docId: DocumentReference? = null)
```

Comment:
When I use data types, I must provide a initial default value (like Java POJO), so it is easier for me to use the nullable types so that I can set the default value to be null.
Under the hood, nullable class and non-nullable class are encoded by different encode functions (`encodeNull(), and encdoeValue()`), so I have to test both cases.

Yes, most code in data classes uses `val`, but for `DocumentId` and `ServerTimestamp`, I need `var` to replace the value with FieldValue and DocumentReference.